packages/manager/modules/pci/src/projects/project/notebooks/notebooks.routing.js

```diff
@@ -14,8 +19,34 @@ export default /* @ngInject */ ($stateProvider) => {
     resolve: {
       goToAddNotebook: /* @ngInject */ ($state, projectId) => () =>
         $state.go('pci.projects.project.notebooks.add', { projectId }),
-      notebooks: /* @ngInject */ ($q, NotebookService, projectId) =>
-        NotebookService.getNotebooks(projectId),
+
+      notebooks: /* @ngInject */ (
+        editors,
+        frameworks,
+        NotebookService,
+        projectId,
+      ) =>
+        NotebookService.getNotebooks(projectId).then((notebooks) =>
+          notebooks.map((notebook) => {
+            set(
+              notebook,
+              'spec.env.editor',
+              find(editors, { id: notebook.spec.env.editorId }),
+            );
+            set(
+              notebook,
+              'spec.env.framework',
+              find(frameworks, { id: notebook.spec.env.frameworkId }),
+            );
```

Comment:
Can we create a new object instead of updating given parameter ?

Reply:
It would have been easier that way, if the parameter to modify is at the top level. Here it is 3 levels deep and the code becomes complex

Reply:
Maybe we could at least copy the object before ?

Reply:
Yeah... Using merge to create a new object without editing the existing one.